foundation-ui/mount/MountService.js

```diff
@@ -0,0 +1,94 @@
+import {EventEmitter} from 'events';
+import {CHANGE} from './MountEvent';
+
+/**
+ * List of component descriptors
+ *
+ * @type {Array.<{component:function, priority:Number, type:String}>}
+ */
+const components = [];
+
+/**
+ * Consecutive number used to rank components based on order of registration
+ *
+ * @type {number}
+ */
+let rank = 0;
+
+/**
+ * MountService
+ */
+class MountService extends EventEmitter {
+
+  /**
+   * Register component with type and optional priority
+   *
+   * respective mounts
+   * @param {function} component - React component which will be
+   * mounted based on the provided type
+   * @param {String} type - String which is used to match components with the
+   * @param {Number} [priority] - Number which is used to sort the components
+   */
+  registerComponent(component, type, priority = 0) {
+    if (typeof component !== 'function') {
+      if (process.env.NODE_ENV !== 'production') {
+        throw new TypeError('Provided component must be a ' +
+            'React.Component constructor or a stateless functional component.');
+      }
+
+      return;
+    }
+
+    if (typeof type !== 'string' || type === '') {
+      if (process.env.NODE_ENV !== 'production') {
+        throw new TypeError('Provided type must be a none empty string');
+      }
+
+      return;
+    }
+
+    // Add component descriptor and sort components list by priority and rank
+    components.push({component, priority, type, rank: rank++});
+    components.sort((a, b) => {
+      if (a.priority !== b.priority) {
+        return b.priority - a.priority;
+      }
+
+      return a.rank - b.rank;
+    });
```

Comment:
The components are sorted in a way that the one with the highest priority comes first and the rest maintains it's position. This enables us to easily limit the components to the ones with the highest priority (e.g. only using the first two components). The latter ensures that we don't have any funny effects (element shuffle) caused by the sorting algorithm.  

### Example

In case one would add the following set of components in the respective order, the result would return a nicely ordered list.

#### Register Component
```
1. B {priority: 0}
2. C {priority: 0}
3. A {priority: 2}
4. D {priority: 0}
```
#### Find Components

```
[A,B,C,D]
```

*The following test verifies that the service returns the components in proper order:*
https://github.com/dcos/dcos-ui/pull/1130/files#diff-ab5a90aae6458cd23f55dae4c16b75b0R135

I'm happy to adjust the sorting if you think that the sorting needs to work differently.